ops/model.py

```diff
@@ -2170,6 +2661,111 @@ def _run(self, *args: str, return_output: bool = False,
                 else:
                     return text
 
+    @staticmethod
+    def _validate_secret_content(content: Optional[Dict[Any, Any]]):
+        if content:
+            for k, v in content.items():
+                if not (isinstance(k, str) and isinstance(v, str)):
+                    raise ValueError('cannot set secret contents to non-string '
+                                     'type {}:{}'.format(type(k).__name__, type(v).__name__))
+
+    def secret_set(self, id: Optional[str],
+                   content: Optional[Dict[str, str]] = None,
+                   description: Optional[str] = None,
+                   label: Optional[str] = None) -> None:
+        self._validate_secret_content(content)
+        args = []  # type: List[str]
+        if id:
+            # Id is optional because we could set a secret by label.
+            args.append(id)
+        if description:
+            args += ['--description', description]
+        if label:
+            args += ['--label', label]
+        if content:
+            for k, v in content.items():
+                args.append('{}="{}"'.format(k, v))
```

Comment:
Unfortunately in no other place in ops we do string validation of that kind. We only recently started to raise exceptions if some key/values are not strings, instead of casting everything silently to str. 
I'm not against it, but it feels out of scope for this PR.

Reply:
Well, the main thing I'm asking for here is that we handle quotes properly. In fact, we're using `subprocess.run`, so we should just be able to not quote the value at all (`subprocess.run` handles that). So I believe we can just do this: `args.append(k+'='+v)`

In addition, let's consider the `bytes` / binary / base64 case (though I'd be happy to do that in the subsequent PR if you prefer).

Reply:
already addressed elsewhere.